tests/cli/test_detect_format.py

```diff
@@ -0,0 +1,113 @@
+from unittest.case import TestCase
+import contextlib
+import io
+import json
+import os
+import os.path as osp
+import shutil
+
+from datumaro.plugins.ade20k2017_format import Ade20k2017Importer
+from datumaro.plugins.ade20k2020_format import Ade20k2020Importer
+from datumaro.plugins.image_dir_format import ImageDirImporter
+from datumaro.util.os_util import suppress_output
+from datumaro.util.test_utils import TestDir
+from datumaro.util.test_utils import run_datum as run
+
+from tests.requirements import Requirements, mark_requirement
+
+ADE20K2017_DIR = osp.join(
+    osp.dirname(__file__), '../assets/ade20k2017_dataset/dataset')
+ADE20K2020_DIR = osp.join(
+    osp.dirname(__file__), '../assets/ade20k2020_dataset/dataset')
+
+class DetectFormatTest(TestCase):
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_unambiguous(self):
+        output_file = io.StringIO()
+
+        with contextlib.redirect_stdout(output_file):
+            run(self, 'detect-format', ADE20K2017_DIR)
+
+        output = output_file.getvalue()
+
+        self.assertIn(Ade20k2017Importer.NAME, output)
+        self.assertNotIn(Ade20k2020Importer.NAME, output)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_ambiguous(self):
+        with TestDir() as test_dir:
+            annotation_dir = osp.join(test_dir, 'training/street')
+            os.makedirs(annotation_dir)
+
+            for asset in [
+                osp.join(ADE20K2017_DIR, 'training/street/1_atr.txt'),
+                osp.join(ADE20K2020_DIR, 'training/street/1.json'),
+            ]:
+                shutil.copy(asset, annotation_dir)
+
+            output_file = io.StringIO()
+
+            with contextlib.redirect_stdout(output_file):
+                run(self, 'detect-format', test_dir)
+
+            output = output_file.getvalue()
+
+            self.assertIn(Ade20k2017Importer.NAME, output)
+            self.assertIn(Ade20k2020Importer.NAME, output)
+
+    # Ideally, there should be a test for the case where no formats match,
+    # but currently that's impossible, because some low-confidence detectors
+    # always match.
```

Comment:
You can add such test on API level. I see, there are no API-level tests though for the new function, so maybe we should introduce them and move some tests there?

Reply:
Yeah, that's a good idea. I'll do that.

Reply:
Well, as it turns out, API-level tests don't really help with this case, because there's no special logic on the API level for the "zero formats detected" case - that logic only exists in the CLI command. However, the API tests could still be useful, so I added them.